x-pack/plugins/synthetics/common/lib/stringify_kueries.ts

```diff
@@ -48,7 +48,7 @@ export const stringifyKueries = (
     }, '');
 };
 
-const isAlphaNumeric = (str: string) => {
+export const isAlphaNumeric = (str: string) => {
```

Comment:
This isn't in scope for your patch but, this function isn't really `isAlphaNumeric`, it's `isNotTheCharactersListedInThisRegexLiteral`. This is a really commonly-solved problem, it might be worth a follow-up refactor to change this to use something like `"^[a-zA-Z0-9_]*$"` from [this SO answer](https://stackoverflow.com/questions/336210/regular-expression-for-alphanumeric-and-underscores), which will explicitly accept alpha/numbers and reject everything else. I think it's a better practice than deny-listing certain characters, unless there's some specific reason we have to do it this way that I am missing.

Reply:
After talking offline there's a well-founded reason for doing it this way, it has to do with the need to sometimes allow `-` for things like `monitor.id: my-id`. I think we can rename this function or supply a comment explaining the implementation to make it clearer why we're denying instead of allowing.